modules/media/jsx/mediaIndex.js

```diff
@@ -1,121 +1,185 @@
+        return <td className={style}><a href={sessionURL}>{cell}</a></td>;
+      }
+      break;
+    case 'Edit Metadata':
+      const editURL = loris.BaseURL + '/media/edit/?id=' + row['Edit Metadata'];
+      return <td className={style}><a href={editURL}>Edit</a></td>;
+      break;
+    }
 
-  resetFilters() {
-    this.refs.mediaFilter.clearFilter();
+    return <td className={style}>{cell}</td>;
   }
 
   render() {
+    // If error occurs, return it.
+    if (this.state.error) {
+      return this.state.error;
+    }
+
     // Waiting for async data to load
     if (!this.state.isLoaded) {
-      return (
-        <Loader/>
-      );
+      return <Loader/>;
     }
 
-    let uploadTab;
-    let tabList = [
-      {id: 'browse', label: 'Browse'},
+    const options = this.state.data.fieldOptions;
+    const fields = [
+      {name: 'pSCID', label: 'PSCID', type: 'text'},
+      {name: 'visitLabel', label: 'Visit Label', type: 'select',
+        options: options.visits},
+      {name: 'instrument', label: 'Instrument', type: 'select',
+        options: options.instruments},
+      {name: 'fileName', label: 'File Name', type: 'text'},
+      {name: 'site', label: 'For Site', type: 'select',
+        options: options.sites},
+      {name: 'language', label: 'Language', type: 'select',
+        options: options.languages},
+      {name: 'fileType', label: 'File Type', type: 'select',
+        options: options.fileTypes},
+      {name: 'uploadedBy', label: 'Uploaded By', type: 'text'},
+      {name: 'hideFile', label: 'File Visibility', type: 'select',
+        options: options.hidden, hide: !this.props.hasPermission('superUser')},
     ];
 
-    if (loris.userHasPermission('media_write')) {
-      tabList.push({id: 'upload', label: 'Upload'});
-      uploadTab = (
-        <TabPane TabId={tabList[1].id}>
-          <MediaUploadForm
-            DataURL={`${loris.BaseURL}/media/ajax/FileUpload.php?action=getData`}
-            action={`${loris.BaseURL}/media/ajax/FileUpload.php?action=upload`}
-            maxUploadSize={this.state.Data.maxUploadSize}
-          />
-        </TabPane>
-      );
-    }
+   /**
+    * XXX: Currently, the order of these headers MUST match the order of the
+    * queried columns in _setupVariables() in media.class.inc
+    */
+    const headers = [
+     {label: 'File Name', display: true},
+     {label: 'PSCID', display: true},
+     {label: 'Visit Label', display: true},
+     {label: 'Language', display: true},
+     {label: 'Instrument', display: true},
+     {label: 'Site', display: true},
+     {label: 'Uploaded By', display: true},
+     {label: 'Date Taken', display: true},
+     {label: 'Comments', display: true},
+     {label: 'Date Uploaded', display: true},
+     {label: 'File Type', display: false},
+     {label: 'Cand ID', display: false},
+     {label: 'Session ID', display: false},
+     {label: 'Hide File', display: false},
+     {label: 'Edit Metadata', display: true},
+    ];
```

Comment:
`fields` and `headers` were placed as separate arrays because that's the way things had previously been setup. However, they don't need to be and can likely be placed in to a single array.